### ISSUE TITLE
For accessibility, app should make use of Dynamic Type font size setting

### ISSUE THREAD

**Issue by gennaios** [nextcloud/ios#2483]
### Steps to reproduce

1. Make any change to Dynamic Type font size in Settings

### Expected behaviour

Nextcloud app should adjust font sizes for table rows and other UX elements as with other apps such as Files iOS or Settings itself.

### Actual behaviour

Tell was what instead happens.

### Screenshots

Included of Files app with a few settings.

![IMG_0125](https://github.com/nextcloud/ios/assets/12172/2dfea5b4-1061-45fe-b9a5-4cbabf01579a)
![IMG_0127](https://github.com/nextcloud/ios/assets/12172/4cb17ba0-7d8e-4509-b087-edd127d687a8)
![IMG_0128](https://github.com/nextcloud/ios/assets/12172/fa16c774-ead7-4758-8dc7-492cf9b2f61a)

### Logs

```
If applicable, you can post the iOS app or server logs (removing any sensitive information).
```

### Reasoning or why should it be changed/implemented?

Larger font sizes perhaps particularly important for those with vision impairment:

**Comment by gennaios** [comment]
https://developer.apple.com/documentation/uikit/uifont/scaling_fonts_automatically

**Comment by VLNickR** [comment]
Yeah, we have a client who is having trouble as the text becomes meshed and unreadable at larger font sizes
